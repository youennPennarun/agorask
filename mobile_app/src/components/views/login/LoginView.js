/* @flow */

import React, {Component} from 'react';
import { View, Text, StyleSheet, Dimensions, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import {MKButton, MKTextField, MKColor} from 'react-native-material-kit';

import {login} from '../../../redux/actions/user';
import {popRoute} from '../../../redux/actions/router';


const {width} = Dimensions.get('window');

export class LoginView extends Component {
  state = {
    username: 'Hello',
    password: '',
  }
  login() {
    const {username, password} = this.state;
    const {doLogin} = this.props;
    if (username && password) {
      doLogin(username, password);
    }
  }

  isFormValid(): boolean {
    return (
      this.state.username !== '' &&
      this.state.password !== ''
    );
  }

  _renderErrorMessage(): ?Object {
    const {message} = this.props;
    if (message) {
      return (
        <Text style={styles.errorMessage}>{message}</Text>
      );
    }
    return null;
  }

  render(): Object {
    const {goToMap} = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.logo} />
          {this._renderErrorMessage()}
          <Textfield style={styles.textInput}
            placeholder='Username'
            blurOnSubmit
            value={this.state.username}
            onChangeText={(text: string) => { this.setState({username: text}); }} />
          <Textfield style={styles.textInput}
            placeholder='Password'
            secureTextEntry
            blurOnSubmit
            value={this.state.password}
            onChangeText={(text: string) => { this.setState({password: text}); }} />

          <LoginButton enabled={this.isFormValid()}
            onPress={() => { this.login(); }} />

          <TouchableOpacity style={styles.goToMapButton}
            onPress={() => { goToMap(); }}>
            <Text>Not Now</Text>
          </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  logo: {
    marginTop: 15,
    width: width - 200,
    height: 150,
    backgroundColor: 'red',
  },
  errorMessage: {
    color: 'red',
  },
  textInput: {
    height: 50,
    width: width - 100,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
  },
  goToMapButton: {
    marginTop: 10,
  },
});
const LoginButton = MKButton.coloredButton()
  .withText('Login')
  .build();

const Textfield = MKTextField.textfieldWithFloatingLabel()
  .withPlaceholder('Search')
  .withUnderlineEnabled(true)
  .withHighlightColor(MKColor.Lime)
  .build();


const mapStateToProps = state => ({
  ...state.user,
});

const mapDispatchToProps = (dispatch) => ({
  doLogin: (username: string, password: string) => {
    dispatch(login(username, password));
  },
  goToMap: (): null => dispatch(popRoute()),
});


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoginView);
