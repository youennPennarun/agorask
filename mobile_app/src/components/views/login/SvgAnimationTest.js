import React, {Component} from 'react';
import {View, Animated, TouchableOpacity} from 'react-native';
import SVG, {Circle} from 'react-native-svg';
import AnimatedImplementation from 'AnimatedImplementation';

const AnimatedSVG = {
  Circle: AnimatedImplementation.createAnimatedComponent(Circle),
};

class SvgAnimationTest extends Component {
  state = {
    radius: new Animated.Value(45),
  }
  animate() {
    Animated.timing(this.state.radius, {
      toValue: 20,
      duration: 2000,
    }).start(() => {
      Animated.timing(this.state.radius, {
      toValue: 45,
      duration: 2000,
    }).start()
    });
  }
  interpolateToString(value: Animated.Value, range: number[]) {
    const outputRange = [`${range[0]}`, `${range[1]}`];
    return value.interpolate({
      inputRange: range,
      outputRange,
    });
  }

  render() {
    return (
      <View>
        <SVG height='100' width='100' >
          <AnimatedSVG.Circle cx='50' cy='50' r={this.interpolateToString(this.state.radius, [20, 45])} fill='green' />
        </SVG>
        
        <TouchableOpacity style={{
            width: 50,
            height: 50,
            backgroundColor: 'blue',
          }}
          onPress={() => { this.animate(); }} />
      </View>
    );
  }
}

export default SvgAnimationTest;
