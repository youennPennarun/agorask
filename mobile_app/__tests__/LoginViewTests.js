/* global jest, expect, describe, it */

import 'react-native';
import React from 'react';

import {LoginView} from '../src/components/views/login/LoginView';

import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

it('renders correctly', () => {
  const wrapper = shallow(<LoginView />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('Button disabled if username is not filled', () => {
  const doLogin = jest.fn();
  const wrapper = shallow(<LoginView doLogin={doLogin} />);
  wrapper.setState({username: '', password: 'Look at me'});
  wrapper.find('BuiltButton').at(0).simulate('press');
  expect(doLogin.mock.calls.length).toBe(0);
});

it('Button disabled if password is not filled', () => {
  const doLogin = jest.fn();
  const wrapper = shallow(<LoginView doLogin={doLogin} />);
  wrapper.setState({username: 'mrMeeseeks', password: ''});
  wrapper.find('BuiltButton').at(0).simulate('press');
  expect(doLogin.mock.calls.length).toBe(0);
});

it('Button enabled if username and password are filled', () => {
  const doLogin = jest.fn();
  const wrapper = shallow(<LoginView doLogin={doLogin} />);
  wrapper.setState({username: 'mrMeeseeks', password: 'Look at me'});
  wrapper.find('BuiltButton').at(0).simulate('press');
  expect(doLogin.mock.calls.length).toBe(1);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it(
  'Should NOT login when submitting the password if the password is not filled',
  () => {
    const doLogin = jest.fn();
    const wrapper = shallow(<LoginView doLogin={doLogin} />);
    wrapper.setState({username: 'mrMeeseeks', password: ''});
    wrapper.find('BuiltTextfield').at(1).props().onSubmitEditing();
    expect(doLogin.mock.calls.length).toBe(0);
  },
);
it(
  'Should login when submitting the password if username and password are filled',
  () => {
    const doLogin = jest.fn();
    const wrapper = shallow(<LoginView doLogin={doLogin} />);
    wrapper.setState({username: 'mrMeeseeks', password: 'Look at me'});
    wrapper.find('BuiltTextfield').at(1).props().onSubmitEditing();
    expect(doLogin.mock.calls.length).toBe(1);
  },
);

it('Should go back on receiving token', () => {
  const back = jest.fn();
  const props = {navigator: {back}};
  const wrapper = shallow(<LoginView {...props} />);
  wrapper.setProps({...props, token: 'token'});
  expect(back.mock.calls.length).toBe(1);
});

it('Should go back owhen clicking on \'Skip login\'', () => {
  const back = jest.fn();
  const props = {navigator: {back}};
  const wrapper = shallow(<LoginView {...props} />);
  wrapper.find('TouchableOpacity').at(0).simulate('press');
  expect(back.mock.calls.length).toBe(1);
});