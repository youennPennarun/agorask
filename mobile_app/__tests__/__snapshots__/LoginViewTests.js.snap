exports[`test Button enable if usrname and password are filled 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 150,
        "marginTop": 15,
        "width": 550,
      }
    } />
  <MKTextField
    blurOnSubmit={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Username"
    style={
      Object {
        "height": 45,
        "marginBottom": 15,
        "width": 650,
      }
    }
    value="mrMeeseeks" />
  <MKTextField
    blurOnSubmit={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    password={true}
    placeholder="Password"
    secureTextEntry={true}
    selectTextOnFocus={true}
    style={
      Object {
        "height": 45,
        "marginBottom": 15,
        "width": 650,
      }
    }
    value="Look at me" />
  <MKButton
    enabled={true}
    onPress={[Function]} />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "marginTop": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Not Now
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 150,
        "marginTop": 15,
        "width": 550,
      }
    } />
  <MKTextField
    blurOnSubmit={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Username"
    style={
      Object {
        "height": 45,
        "marginBottom": 15,
        "width": 650,
      }
    }
    value="" />
  <MKTextField
    blurOnSubmit={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    password={true}
    placeholder="Password"
    secureTextEntry={true}
    selectTextOnFocus={true}
    style={
      Object {
        "height": 45,
        "marginBottom": 15,
        "width": 650,
      }
    }
    value="" />
  <MKButton
    enabled={false}
    onPress={[Function]} />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "marginTop": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Not Now
    </Text>
  </TouchableOpacity>
</View>
`;
